### DISCUSSION TITLE
How can I autoFocus / pass focus from the Menu to a child element?

### DISCUSSION THREAD

**Discussion by httpete** [ariakit/ariakit#2471] [Questions] [answered]
I have a menu that contains some form fields, and upon opening the menu I want the first form field to get focus. Any ideas? I am trying passing ref, and autoFocus, and trying to focus on menu open.

**Comment by httpete** [comment, marked as answer]
I got this - and it's marvelous. If anyone else hits this and wants to know how:

You really can't isolate focus behavior without doing this:

https://developer.chrome.com/docs/devtools/accessibility/focus/

If you are in storybook, you have to choose the right iframe to get it to work
![image](https://github.com/ariakit/ariakit/assets/1200635/f8a4696d-97b8-47c9-b599-8f38743d4328)

By passing composite={false} and setting initialFocus to a ref like so, it works. But it is a little odd, the autoFocus doesn't seem to work on the child element, only setting initialFocus to an empty ref does the job?


```

  const selectRef = useRef<HTMLElement>(null);

      <Menu
        store={menu}
        as={Paper}
        sx={{ mt: 1, ml: '-10px', p: 1 }}
        composite={false}
        initialFocus={selectRef}

**Reply by httpete** [reply]
@diegohaz isn't it supposed to auto focus the first element by default? It won't for me without passing the empty ref...

**Reply by diegohaz** [reply]
I'm not sure I understand your use case. It looks like you're using `Menu` when you actually want `Popover`. Could you share some screenshots/videos of that UI so I can have a better sense?

**Reply by httpete** [reply]
You are correct - I was using a Menu where I should have used a Popover. It felt menu-ish to me since it has checkboxes but we have cancel / apply buttons so it is really just a PopoverDialog idea. Because the ariakit api is so parallel it was easy for me to use that, and I'm done! thanks!!
